experimental/casemap/src/titlecase.rs

```diff
@@ -0,0 +1,401 @@
+    pub const fn new_with_mapper(casemapper: CM) -> Self {
+        Self {
+            cm: casemapper,
+            gc: Some(DataPayload::from_static_ref(
+                icu_properties::provider::Baked::SINGLETON_PROPS_GC_V1,
+            )),
+        }
+    }
+    /// A constructor which creates a [`TitlecaseMapper`] from an existing [`CaseMapper`]
+    /// (either owned or as a reference), with the legacy head adjustment behavior.
+    /// See struct docs on [`TitlecaseMapper`] for more information on head adjustment behavior.
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    pub const fn new_with_mapper_legacy(casemapper: CM) -> Self {
+        Self {
+            cm: casemapper,
+            gc: None,
+        }
+    }
+    /// Construct this object to wrap an existing CaseMapper (or a reference to one), loading additional data as needed.
+    ///
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::new_with_mapper)]
+    pub fn try_new_with_mapper_unstable<P>(provider: &P, casemapper: CM) -> Result<Self, DataError>
+    where
+        P: DataProvider<CaseMapV1Marker> + DataProvider<GeneralCategoryV1Marker> + ?Sized,
+    {
+        let gc = provider.load(Default::default())?.take_payload()?;
+        Ok(Self {
+            cm: casemapper,
+            gc: Some(gc),
+        })
+    }
+
+    /// Returns the full titlecase mapping of the given string as a [`Writeable`], treating
+    /// the string as a single segment (and thus only titlecasing the beginning of it).
+    ///
+    /// This should typically be used as a lower-level helper to construct the titlecasing operation desired
+    /// by the application, for example one can titlecase on a per-word basis by mixing this with
+    /// a `WordSegmenter`.
+    ///
+    /// This function is context and language sensitive. Callers should pass the text's language
+    /// as a `LanguageIdentifier` (usually the `id` field of the `Locale`) if available, or
+    /// `Default::default()` for the root locale.
+    ///
+    /// See [`Self::titlecase_segment_to_string()`] for the equivalent convenience function that returns a String,
+    /// as well as for an example.
+    pub fn titlecase_segment<'a>(
+        &'a self,
+        src: &'a str,
+        langid: &LanguageIdentifier,
+        options: TitlecaseOptions,
+    ) -> impl Writeable + 'a {
+        if let Some(gc) = self.gc.as_ref() {
+            // letter, number, symbol, or private use code point
+            const HEAD_GROUPS: GeneralCategoryGroup = GeneralCategoryGroup::Letter
+                .union(GeneralCategoryGroup::Number)
+                .union(GeneralCategoryGroup::Symbol)
+                .union(GeneralCategoryGroup::PrivateUse);
+            let gc = gc.get();
+            self.cm
+                .as_ref()
+                .titlecase_segment_with_adjustment(src, langid, options, |_data, ch| {
+                    HEAD_GROUPS.contains(gc.get32(ch as u32))
```

Comment:
Question: Why are you using the private APIs on GC instead of the public onces?

Reply:
See above, the public ones do not work if you want to share code between baked and loaded modes.

Reply:
fixed